worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -127,6 +129,13 @@ public PartitionFilesSorter(
         ThreadUtils.newDaemonCachedThreadPool(
             "worker-file-sorter-execute", conf.partitionSorterThreads(), 120);
 
+    indexCache =
+        CacheBuilder.newBuilder()
+            .concurrencyLevel(conf.partitionSorterThreads())
+            .expireAfterAccess(conf.partitionSorterIndexExpire(), TimeUnit.MILLISECONDS)
+            .maximumSize(indexCacheSize)
```

Comment:
how about using maximumWeight() instead and number of ShuffleBlockInfos as weight?

```suggestion
            .maximumWeight(indexCacheSize)
            .weigher(new Weigher<String, Map<Integer, List<ShuffleBlockInfo>>>() {
              @Override
              public int weigh(String key, Map<Integer, List<ShuffleBlockInfo>> infoMap) {
                return infoMap.values().stream().mapToInt(List::size).sum();
              }
            })
```